### ISSUE TITLE
PM2-B

### ISSUE THREAD

**Issue by alexanderquispe** [alexanderquispe/14.38_Causal_ML#30]
1. Inference on Predictive and Causal Effects in High Dimensional Linear Regression Models

**Comment by alexanderquispe** [comment]
* pm2_notebook_jannis
In this JN we use the double Lassso regression. Again we face the problem of equivalent "alphas" 
I used alpha = 0.00077 (manually chosen)  as the best proxy to get similar results as in the RN
Coefficients are similar but CI are not equal.

**Comment by alexanderquispe** [comment]
*python-notebook-experiment-on-orthogonal-learning
I found that the iteration process may have an error. 
![image](https://user-images.githubusercontent.com/27368860/114589891-bc453880-9c4d-11eb-9aab-23d6ee3cd933.png)

I think that this line:
 if sum(SX_IDs) == 0 : 
        Naive[ 0 ] = sm.OLS( Y , sm.add_constant(D) ).fit().summary2().tables[1].round(3).iloc[ 1, 0 ]
should be replaced by
 if sum(SX_IDs) == 0 : 
        Naive[ i ] = sm.OLS( Y , sm.add_constant(D) ).fit().summary2().tables[1].round(3).iloc[ 1, 0 ]

Otherwise, the iteration is not saving the results for that regression. 
Also when I checked the results of the naive matrix you got and the naive matrix from the RNotebook the mean is totally different. But when I change this the results seem to converge. 
@anzonyqr , let me know if this makes sense to you. 
Thanks!

**Comment by alexanderquispe** [comment]
This is what I found in the R code : 
![image](https://user-images.githubusercontent.com/27368860/114591888-f6afd500-9c4f-11eb-8277-84c0b83332b1.png)
This is what we have in the Python code:
![image](https://user-images.githubusercontent.com/27368860/114591942-06c7b480-9c50-11eb-8fc1-1e69036da5ab.png)

I will divide by 4 the error term. And also modify the iterator as I did in the code above.

**Comment by anzonyqr** [comment]
> *python-notebook-experiment-on-orthogonal-learning
> I found that the iteration process may have an error.
> ![image](https://user-images.githubusercontent.com/27368860/114589891-bc453880-9c4d-11eb-9aab-23d6ee3cd933.png)
> 
> I think that this line:
> if sum(SX_IDs) == 0 :
> Naive[ 0 ] = sm.OLS( Y , sm.add_constant(D) ).fit().summary2().tables[1].round(3).iloc[ 1, 0 ]
> 
> **should be replaced by**
> 
> if sum(SX_IDs) == 0 :
> Naive[ i ] = sm.OLS( Y , sm.add_constant(D) ).fit().summary2().tables[1].round(3).iloc[ 1, 0 ]
> 
> Otherwise, the iteration is not saving the results for that regression.
> Also when I checked the results of the naive matrix you got and the naive matrix from the RNotebook the mean is totally different. But when I change this the results seem to converge.
> @anzonyqr , let me know if this makes sense to you.
> Thanks!

You are right. I made a mistake. Zero should be change for *i* otherwise it will not change.

**Comment by anzonyqr** [comment]
> This is what I found in the R code :
> ![image](https://user-images.githubusercontent.com/27368860/114591888-f6afd500-9c4f-11eb-8277-84c0b83332b1.png)
> This is what we have in the Python code:
> ![image](https://user-images.githubusercontent.com/27368860/114591942-06c7b480-9c50-11eb-8fc1-1e69036da5ab.png)
> 
> I will divide by 4 the error term. And also modify the iterator as I did in the code above.

You are right. I have already made corrections. Now, the python script is equal to r code.

**Comment by SandraMartinezGutierrez** [comment]
> 
[JULIA SCRIPT]
When running OLS regression, there is a problem with the intercepts due to the Cholesky factorization. 
![image](https://user-images.githubusercontent.com/81896095/157118028-54701006-088d-4dfc-b26c-7528a4b190a7.png)

Found a solution here: 
1) https://github.com/JuliaStats/StatsModels.jl/issues/31
2) https://github.com/JuliaStats/GLM.jl/issues/426

Final output: 
![image](https://user-images.githubusercontent.com/81896095/157118409-027c55bb-2c8a-48bd-b6ac-a1981fbf22a3.png)

Notice that to iterate over columns from an specific data, I used: 

term(:y) ~  sum(term.(names(data[!, Not(["y", "intercept"])]))

**Comment by SandraMartinezGutierrez** [comment]
[JULIA SCRIPT]

In this section of the Python script, I was looking for a function like .set_index() in python at Julia dataframe.
![image](https://user-images.githubusercontent.com/81896095/157119047-2aad19c1-d175-4f0c-9cf2-4cbd2a64eb3d.png)

I've found out NamedArray can give similar result with .set_index() in Python as below:

![image](https://user-images.githubusercontent.com/81896095/157119200-91a5ee39-6c96-4aab-9208-421e5890db29.png)

However, NamedArray returns as matrix format only, and I could not find a function to be used in Julia dataframe. For this reason, it is recommended to add the "index information" as a column in the dataframe. Then, it is possible to make a groupby to the data by that column to have a quick lookup.

Documentation about NamedArrays can be found here: 
1) https://github.com/davidavdav/NamedArrays.jl

Final output: 
![image](https://user-images.githubusercontent.com/81896095/157119700-fd7e6693-dd47-4497-a835-4c65e606bb65.png)

**Comment by SandraMartinezGutierrez** [comment]
[JULIA SCRIPT]

In the Python script, when trying to convert this table to HTM, it vas only necessary to add "to_html()".

![image](https://user-images.githubusercontent.com/81896095/157120040-8dab91dd-abee-4ec3-897e-77b4a7903a05.png)

However, in Julia, I found out an interesting way to convert tables to HTML.

For more information: 
1) https://ronisbr.github.io/PrettyTables.jl/stable/
2) https://github.com/ronisbr/PrettyTables.jl

Final output:
![image](https://user-images.githubusercontent.com/81896095/157120690-21b46ea7-e21a-4c25-b5bb-d05591ef8129.png)

**Comment by SandraMartinezGutierrez** [comment]
> * pm2_notebook_jannis
>   In this JN we use the double Lassso regression. Again we face the problem of equivalent "alphas"
>   I used alpha = 0.00077 (manually chosen)  as the best proxy to get similar results as in the RN
>   Coefficients are similar but CI are not equal.

[JULIA SCRIPT]

Julia_pm2_notebook_jannis
In this notebook, when using the double Lassso regression, we face the problem of equivalent "alphas"
I used alpha = 0..8 (manually chosen)  as the best proxy to get similar results as in the RN
Coefficients are similar but CI are not equal.


R Notebook::
![image](https://user-images.githubusercontent.com/81896095/157186596-477b49f2-4e72-4dce-afc5-3811abafdbf8.png)


Julia Notebook: 
![image](https://user-images.githubusercontent.com/81896095/157187796-1419fe27-febd-4ec0-a3ce-c2579a888bb6.png)
